core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -550,8 +550,7 @@ private Optional<NormalizedSimpleComparison> toNormalizedSimpleComparison(Compar
             Type leftType = expressionTypes.get(NodeRef.of(comparison.getLeft()));
             Type rightType = expressionTypes.get(NodeRef.of(comparison.getRight()));
 
-            // TODO: re-enable this check once we fix the type coercions in the optimizers
```

Comment:
when this was fixed...?

Reply:
This PR fixes some tests. I don't know whether there were any other fixes related to this.

Reply:
Sure, but this change is not in tests, but in product

Reply:
Agreed. The check is sound. The tests say it should be fine. What else can I do?

Reply:
Why was check disabled in the first place and what fixed it so that we can enable it again? Was it only tests?

Reply:
The check was introduced by @losipiuk in 03f594265f504f6e7b63f832e39902e509e95220
and disabled by @erichwang in a103a639e396608d6e75f68d0b039597cd352f35.
Eric made it clear that the check is disabled temporarily.

The check is sound and important for reasoning about optimizers, we rely on comparison's sides being same type in many places. We shouldn't be afraid of enabling the check here.

Reply:
Sure, but what changed so that https://github.com/trinodb/trino/commit/a103a639e396608d6e75f68d0b039597cd352f35 is no longer relevant?

Reply:
it was over 5 years ago. i think it will be hard to find out.

Reply:
Does it have potential to cause regressions?

Reply:
Absolutely! (as pretty much any other change in the production code base)